framework/wazuh/core/utils.py

```diff
@@ -145,21 +145,31 @@ def process_array(array: list, search_text: str = None, complementary_search: bo
         List of fields allowed to select from.
     filters : dict
         Defines required field filters. Format: {"field1":"value1", "field2":["value2","value3"]}
+    distinct : bool
+        Look for distinct values.
 
     Returns
     -------
     dict
         Dictionary: {'items': Processed array, 'totalItems': Number of items, before applying offset and limit)}
     """
     if not array:
-        return {'items': list(), 'totalItems': 0}
+        return {'items': [], 'totalItems': 0}
 
-    if isinstance(filters, dict) and len(filters.keys()) > 0:
-        new_array = list()
+    filter_items = isinstance(filters, dict) and len(filters.keys()) > 0
+    if distinct or filter_items:
+        new_array = []
         for element in array:
-            for key, value in filters.items():
-                if element[key] in value:
-                    new_array.append(element)
+            if distinct and (element in new_array):
+                continue
```

Comment:
`distinct` should probably be applied after `select`. However, `select` should not be moved according to:
- https://github.com/wazuh/wazuh/issues/13411#issuecomment-1122592760